addons/wowl/static/src/components/dropdown/dropdown.ts

```diff
@@ -51,8 +55,11 @@ export class Dropdown extends Component {
     this._changeStateAndNotify({ open: true, groupIsOpen: true });
   }
 
-  private _toggle() {
+  private async _toggle() {
     const toggled = !this.state.open;
+    if (toggled && this.props.prepareOpenDropDown) {
+      await this.props.prepareOpenDropDown();
```

Comment:
Await ?
It depends of what behavior we want, but awaiting for an outside function may delay the opening of the dropdown.

Have you considered emitting a custom event, say 'beforeOpen' ?
That would allow the dropdown user to display some kind of loading spinner while fetching some content.

@kebeclibre  @FrancoisGe

Reply:
prop name: beforeOpen